### PR TITLE
Removed dependency on Inference Engine from transformation library

### PR DESCRIPTION
**Pull request by ilyachur**
Introduced export macros for transformation library

### CODE REVIEW

File: inference-engine/src/transformations/CMakeLists.txt [hunk 1]
```diff
@@ -23,15 +23,12 @@ source_group("include" FILES ${PUBLIC_HEADERS})
 
 add_library(${TARGET_NAME} SHARED ${LIBRARY_SRC} ${PUBLIC_HEADERS})
 
-target_compile_definitions(${TARGET_NAME} PRIVATE IMPLEMENT_INFERENCE_ENGINE_API)
+target_compile_definitions(${TARGET_NAME} PRIVATE TRANSFORMATIONS_DLL_EXPORTS)
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/CMakeLists.txt:26]: we can reuse `inference_engine_transformations_EXPORTS` which is defined automatically by cmake for dynamic libraries.

Comment (ilyachur) [reply inference-engine/src/transformations/CMakeLists.txt:26]: Ok, will try to change

File: inference-engine/src/transformations/CMakeLists.txt [hunk 2]
```diff
@@ -23,15 +23,12 @@ source_group("include" FILES ${PUBLIC_HEADERS})
 
 add_library(${TARGET_NAME} SHARED ${LIBRARY_SRC} ${PUBLIC_HEADERS})
 
-target_compile_definitions(${TARGET_NAME} PRIVATE IMPLEMENT_INFERENCE_ENGINE_API)
+target_compile_definitions(${TARGET_NAME} PRIVATE inference_engine_transformations_EXPORTS)
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/CMakeLists.txt:26]: we don't have to define it. It's automatically defined as `<shared library name>_EXPORTS`

Comment (ilyachur) [reply inference-engine/src/transformations/CMakeLists.txt]: Got it. Fixed.

**Comment by GlebKazantaev** [comment]
@ilyachur lets rebase it to the latest master and then I'll merge this PR.

**Pull request merged**